packages/auth0_api_python/src/auth0_api_python/api_client.py

```diff
@@ -14,46 +29,210 @@ class ApiClient:
-            jwks_uri = metadata["jwks_uri"]
-            self._jwks_data = await fetch_jwks(
-                jwks_uri=jwks_uri, 
-                custom_fetch=self.options.custom_fetch
+
+    async def verify_request(
+        self,
+        headers: dict[str, str],
+        http_method: Optional[str] = None,
+        http_url: Optional[str] = None
+    ) -> dict[str, Any]:
+        """
+        Dispatch based on Authorization scheme:
+          • If scheme is 'DPoP', verifies both access token and DPoP proof
+          • If scheme is 'Bearer', verifies only the access token
+
+        Args:
+            headers: HTTP headers dict containing:
+                - "authorization": The Authorization header value (required)
+                - "dpop": The DPoP proof header value (required for DPoP)
+            http_method: The HTTP method (required for DPoP)
+            http_url: The HTTP URL (required for DPoP)
+
+        Returns:
+            The decoded access token claims
+
+        Raises:
+            MissingRequiredArgumentError: If required args are missing
+            InvalidAuthSchemeError: If an unsupported scheme is provided
+            InvalidDpopProofError: If DPoP verification fails
+            VerifyAccessTokenError: If access token verification fails
+        """
+        authorization_header = headers.get("authorization", "")
+        dpop_proof = headers.get("dpop")
+
+        if not authorization_header:
+            if getattr(self.options, "dpop_required", False):
+                raise self._prepare_error(
+                        InvalidAuthSchemeError("Expecting Authorization header with DPoP scheme.")
+                    )
+            else :
+                raise self._prepare_error(MissingAuthorizationError())
+
+
+        parts = authorization_header.split(" ")
+        if len(parts) != 2:
+            if len(parts) < 2:
+                raise self._prepare_error(MissingAuthorizationError())
+            elif len(parts) > 2:
+                raise self._prepare_error(
+                    InvalidAuthSchemeError("Invalid Authorization HTTP Header Format for authorization")
+                )
+
+
+        try:
+            scheme, token = authorization_header.split(" ", 1)
+        except ValueError:
+            raise self._prepare_error(
+                MissingAuthorizationError()
             )
-        return self._jwks_data
+
+
+        scheme = scheme.strip().lower()
+
+        if getattr(self.options, "dpop_required", False) and scheme != "dpop":
+                if scheme == "bearer":
+                    raise self._prepare_error(
+                        InvalidAuthSchemeError("Invalid scheme. Expected 'DPoP', but got 'bearer'."),
+                        auth_scheme=scheme
+                    )
+                else:
+                    raise self._prepare_error(
+                        InvalidAuthSchemeError("Invalid scheme. Expected 'DPoP' scheme."),
+                        auth_scheme=scheme
```

Comment:
Could this be simplified by raising an error once and interpolating the `scheme` in the error message (`but got <SCHEME>.`)?